docs/getting-started/queries/gosdk/complex-data-query.md

```diff
@@ -1,5 +1,1312 @@
 ---
 id: complex-data-query
-title: Query Data Using JSON Fields
+title: Query Data Using Fields in JSON value
 ---
 
+Once a database session is created, we can call `session.JSONQuery()` to get `JSONQuery` object.
+
+```go
+// JSONQuery provides method to execute json query on a given user database
+// The query is a json string which must contain predicates under the field
+// selector. The first field in the selector can be a combinational operator
+// such as "$and" or "$or" followed by a list of attributes and a list of
+// conditions per attributes. A query example is shown below
+//
+// {
+//   "selector": {
+// 		"$and": {            -- top level combinational operator
+// 			"attr1": {          -- a field in the json document
+// 				"$gte": "a",    -- value criteria for the field
+// 				"$lt": "b"      -- value criteria for the field
+// 			},
+// 			"attr2": {          -- a field in the json document
+// 				"$eq": true     -- value criteria for the field
+// 			},
+// 			"attr3": {          -- a field in the json document
+// 				"$lt": "a2"     -- a field in the json document
+// 			}
+// 		}
+//   }
+// }
+type JSONQuery interface {
+	Execute(dbName, query string) ([]*types.KVWithMetadata, error)
+}
+```
+
+To query a data, we need to call `Execute()` method of the object from above.
```

Comment:
To query data 
-> data is plural, datum is singular, to query a record to query a value but to query data

this error repeats in several places

Reply:
I did a search using `ack` to find all occurrences. I think `a data transaction` is fine.

Reply:
'a data transaction' is fine because transaction is the subject and is singular.